app/js/components/ArtifactGenerator.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+const style = { verticalAlign: 'middle' };
+
+const ArtifactGenerator = ({
+    active,
+    toggleCreation,
+    createArtifact,
+    selectDirectory,
+    dirPath
+}) => (
+    <table className="table">
+        <thead>
+            <tr>
+                <th className="col-xs-6"></th>
+                <th className="col-xs-4"></th>
+                <th className="col-xs-1"></th>
+            </tr>
+        </thead>
+        <tbody>
+    { active === 0 ?
+        <tr>
+            <td></td>
+            <td></td>
+            <td><button
+                type="button"
+                className="close"
+                aria-label="Create Artifact"
+                onClick={() => toggleCreation(active)}
+            >
+                <span
+                    className="glyphicon glyphicon-plus"
+                    aria-hidden="true"
+                ></span>
+            </button></td>
+        </tr> :
+        <tr>
+            <td style={style}>
+                <div className="input-group">
+                    <span className="input-group-btn">
+                        <button
+                            type="button"
+                            className="btn btn-primary pull-left"
+                            onClick={selectDirectory}
+                        >
+                            <span
+                                className="glyphicon glyphicon-import"
+                            ></span>
+                            <span> Import</span>
+                        </button>
+                    </span>
+                    <input
+                        type="text"
+                        disabled="true"
+                        className="form-control"
+                        style={{ width: '100%' }}
+                        placeholder="Data Directory Path"
+                        value={dirPath || ''}
+                    />
+                </div>
+            </td>
```

Comment:
This isn't actually inside of the `<form>`, so it's never being looked up or referenced. The path is being taken out of state and that `dirPath` variable is just being passed to the `onSubmit` function. So, it's basically just for looks and to show the user what they've selected.